timdex_dataset_api/config.py

```diff
@@ -3,29 +3,22 @@
 
 
 def configure_logger(name: str) -> logging.Logger:
-    """Prepares and returns a logger instance for a given module name.
+    """Prepares a logger instance.
 
-    This approach is suitable for an installed and imported library such as this, where
-    any calling application logging levels and handlers should be utilized.
+    If the env var TDA_LOG_LEVEL is set, the logging level will override the logging
+    level of the calling context.
 
     Args:
         name (str): The name of the logger, typically __name__ is passed by caller
     """
     logger = logging.getLogger(name)
-    logger.addHandler(logging.NullHandler())
 
-    log_level = os.getenv("TDA_LOG_LEVEL", "INFO").strip().upper()
-    if log_level not in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]:
-        raise ValueError(f"Invalid log level: '{log_level}'")
-    logger.setLevel(getattr(logging, log_level))
-
-    handler = logging.StreamHandler()
-    handler.setFormatter(
-        logging.Formatter(
-            "%(asctime)s %(levelname)s %(name)s.%(funcName)s() "
-            "line %(lineno)d: %(message)s"
-        )
-    )
-    logger.addHandler(handler)
+    # set logger level if env var 'TDA_LOG_LEVEL' is set
+    log_level = os.getenv("TDA_LOG_LEVEL")
+    if log_level:
```

Comment:
Could we use the walrus operator here? `if log_level := os.getenv("TDA_LOG_LEVEL")`

Reply:
I dig it!  Good call.